vkcs/backup/backup_shared.go

```diff
@@ -0,0 +1,273 @@
+package backup
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/clients"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/backup/v1/plans"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/backup/v1/providers"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/backup/v1/triggers"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/db/v1/instances"
+)
+
+const (
+	ProviderNameNova  = "cloud_servers"
+	ProviderNameTrove = "dbaase"
+	Nova              = "OS::Nova"
+	Trove             = "OS::Trove"
+	NovaInstance      = "OS::Nova::Server"
+	TroveInstance     = "OS::Trove::Instance"
+
+	RetentionFull = "max_backups"
+	RetentionGFS  = "gfs"
+
+	TimeFormat = "15:04"
+)
+
+var providerNameMapping = map[string]string{
+	"OS::Nova":  ProviderNameNova,
+	"OS::Trove": ProviderNameTrove,
+}
+
+func getResourcesInfo(config clients.Config, region string, instancesID []types.String, resourceType string) ([]*plans.BackupPlanResource, error) {
+	if resourceType == ProviderNameNova {
+		computeClient, err := config.ComputeV2Client(region)
+		if err != nil {
+			return nil, fmt.Errorf("error creating VKCS compute client: %s", err.Error())
+		}
+
+		allPages, err := servers.List(computeClient, servers.ListOpts{}).AllPages()
+		if err != nil {
+			return nil, fmt.Errorf("error getting servers info: %s", err)
+		}
+
+		allServers, err := servers.ExtractServers(allPages)
+		if err != nil {
+			return nil, fmt.Errorf("error getting servers info: %s", err)
+		}
+
+		resourcesInfo := make([]*plans.BackupPlanResource, 0)
+		for _, instanceID := range instancesID {
+			for _, server := range allServers {
+				if instanceID.ValueString() == server.ID {
+					resourceInfo := plans.BackupPlanResource{
+						ID:   server.ID,
+						Type: NovaInstance,
+						Name: server.Name,
+					}
+					resourcesInfo = append(resourcesInfo, &resourceInfo)
+					break
+				}
+			}
+		}
+		if len(resourcesInfo) != len(instancesID) {
+			return nil, fmt.Errorf("error getting resources info: some resources were not found")
+		}
+
+		return resourcesInfo, nil
+	}
+
+	if resourceType == ProviderNameTrove {
+		dbClient, err := config.DatabaseV1Client(region)
+		if err != nil {
+			return nil, fmt.Errorf("error creating VKCS database client: %s", err.Error())
+		}
+
+		allPages, err := instances.List(dbClient).AllPages()
```

Comment:
does cloud backup not work with db clusters?

Reply:
Added cluster support